quickwit-metastore/src/metastore/postgresql_metastore.rs

```diff
@@ -0,0 +1,1139 @@
+                        .map(|deleted_split| deleted_split.split_id.clone())
+                        .any(|x| x == *split_id)
+                    {
+                        error_split_id = split_id;
+                        break;
+                    }
+                }
+                return Err(diesel::result::Error::NotFound);
+            }
+
+            Ok(())
+        })
+        .map_err(|err| match err {
+            diesel::result::Error::NotFound => MetastoreError::SplitDoesNotExist {
+                split_id: error_split_id.to_string(),
+            },
+            _ => MetastoreError::InternalError {
+                message: "Failed to mark splits as deleted".to_string(),
+                cause: anyhow::anyhow!(err),
+            },
+        })?;
+
+        Ok(())
+    }
+
+    async fn index_metadata(&self, index_id: &str) -> MetastoreResult<IndexMetadata> {
+        let conn = self
+            .connection_pool
+            .get()
+            .map_err(|err| MetastoreError::InternalError {
+                message: "Failed to get connection".to_string(),
+                cause: anyhow::anyhow!(err),
+            })?;
+
+        let select_index_statement =
+            schema::indexes::dsl::indexes.filter(schema::indexes::dsl::index_id.eq(index_id));
+        debug!(sql=%debug_query::<Pg, _>(&select_index_statement).to_string());
+        let model_index = select_index_statement
+            .first::<model::Index>(&conn)
+            .map_err(|err| match err {
+                diesel::result::Error::NotFound => MetastoreError::IndexDoesNotExist {
+                    index_id: index_id.to_string(),
+                },
+                _ => MetastoreError::InternalError {
+                    message: "Failed to select index".to_string(),
+                    cause: anyhow::anyhow!(err),
+                },
+            })?;
+
+        let index_metadata =
+            model_index
+                .make_index_metadata()
+                .map_err(|err| MetastoreError::InternalError {
+                    message: "Failed to make index metadata".to_string(),
+                    cause: anyhow::anyhow!(err),
+                })?;
+
+        Ok(index_metadata)
+    }
+
+    fn uri(&self) -> String {
+        self.uri.clone()
+    }
+}
+
+/// A single file metastore factory
+#[derive(Clone)]
+pub struct PostgresqlMetastoreFactory {}
+
+impl Default for PostgresqlMetastoreFactory {
+    fn default() -> Self {
+        PostgresqlMetastoreFactory {}
+    }
+}
+
+#[async_trait]
+impl MetastoreFactory for PostgresqlMetastoreFactory {
+    async fn resolve(&self, uri: &str) -> Result<Arc<dyn Metastore>, MetastoreResolverError> {
+        let metastore = get_postgresql_metastore(uri).await;
+
+        Ok(Arc::new((*metastore).clone()))
+    }
+}
+
+#[cfg(test)]
+#[async_trait]
+impl crate::tests::test_suite::DefaultForTest for PostgresqlMetastore {
+    async fn default_for_test() -> Self {
+        use std::env;
```

Comment:
Can you put instruction somewhere about how to run tests locally?
Typically a docker compose file spinning a postgres and environment variable would be great.

dotenv with a committed .env file pointing at the docker-compose instance is ok, but please make the dotenv a dev-dependency.